index.bs

```diff
@@ -316,56 +317,83 @@ Interface {#interface}
+
+On getting, the {{NavigatorUAData/uaList}} attribute MUST run these steps:
+
+1. Let |list| be a [=/list=].
+
+2. Collect pairs of [=user agent/brand=] and [=user agent/significant version=] which represent the user agent,
+    its equivalence class and/or its rendering engine.
+
+3. For each pair:
+
+    1. Let |dict| be a new {{NavigatorUABrandVersion}} dictionary,
+        with [=user agent/brand=] as {{NavigatorUABrandVersion/brand}} and [=user agent/significant version=] as {{NavigatorUABrandVersion/version}}. 
+
+    2. Append |dict| to |list|.
+
+4.  The user agent MAY execute the following steps:
+
+    1.  [=list/Append=] additional items to |list| containing {{NavigatorUABrandVersion}} objects,
+        initialized with arbitrary {{NavigatorUABrandVersion/brand}} and {{NavigatorUABrandVersion/version}} combinations.
+
+    2.  Randomize the order of the items in |list|.
+
+    Note: See [[#grease]] for more details on why these steps might be appropriate.
+
+5. Return |list|.
+
+On getting, the {{NavigatorUAData/mobile}} attribute must return the user agent's [=user agent/mobileness=].
+
+The <dfn method for="NavigatorUA"><code>getHighEntropyValues()</code></dfn> method, called with [=sequence<DOMString>=] |hints|, MUST run these steps:
 
 1. Let |p| be a [=a new promise=].
 
 2.  Run the following steps [=in parallel=]:
 
-    1.  Let |UAData| be a new {{NavigatorUAData}} object whose values are initialized as follows:
+    1. Let |uaData| be a new {{UADataValues}}.
 
-        :   {{NavigatorUAData/brand}}
-        ::  The user agent's [=user agent/brand=].
-        :   {{NavigatorUAData/getPlatform}}
-        ::  The user agent's [=user agent/platform brand=].
-        :   {{NavigatorUAData/getPlatformVersion}}
-        ::  The user agent's [=user agent/platform version=].
-        :   {{NavigatorUAData/getArchitecture}}
-        ::  The user agent's [=user agent/platform architecture=].
-        :   {{NavigatorUAData/getModel}}
-        ::  The user agent's [=user agent/model=].
-        :   {{NavigatorUAData/mobile}}
-        ::  The user agent's [=user agent/mobileness=].
+    2. If |hints| [=contains=] "platform", let |uaData|["{{UADataValues/platform}}"] be the the user agent's [=user agent/platform brand=].
 
-    2.  [=Resolve=] |p| with |UAData|.
+    3. If |hints| [=contains=] "platformVersion", let |uaData|["{{UADataValues/platformVersion}}"] be the the user agent's [=user agent/platform version=].
 
-3.  Return |p|.
+    4. If |hints| [=contains=] "architecture", let |uaData|["{{UADataValues/architecture}}"] be the the user agent's [=user agent/platform architecture=].
 
-ISSUE: Provide a method to only access the UA's significant version.
+    5. If |hints| [=contains=] "model", let |uaData|["{{UADataValues/model}}"] be the the user agent's [=user agent/model=].
+
+    6. [=Queue a task=] to [=resolve=] |p| with |uaData|.
+
+ISSUE: Add a specific task source this is queued on.
```

Comment:
Thoughts on the task queue for this?

Reply:
What do you plan to implement in Chrome?

There's always the option of giving maximum flexibility by creating an entirely new task source.

The closest counterpart I can think of off the top of my head is createImageBitmap. Unfortunately that doesn't do this right either :(. https://github.com/whatwg/html/issues/5329